sdk/core/Azure.Core/src/Shared/ARMChallengeAuthenticationPolicy.cs

```diff
@@ -13,35 +14,58 @@ namespace Azure.Core.Pipeline
     /// A policy that sends an <see cref="AccessToken"/> provided by a <see cref="TokenCredential"/> as an Authentication header.
     /// Note: This class is currently in preview and is therefore subject to possible future breaking changes.
     /// </summary>
-    internal class ARMChallengeAuthenticationPolicy : BearerTokenChallengeAuthenticationPolicy
+    internal class ARMChallengeAuthenticationPolicy : BearerTokenAuthenticationPolicy
     {
+        private readonly string[] _scopes;
+
         /// <summary>
         /// Creates a new instance of <see cref="ARMChallengeAuthenticationPolicy"/> using provided token credential and scope to authenticate for.
         /// </summary>
         /// <param name="credential">The token credential to use for authentication.</param>
         /// <param name="scope">The scope to authenticate for.</param>
-        public ARMChallengeAuthenticationPolicy(TokenCredential credential, string scope) : base(credential, new[] { scope }) { }
+        public ARMChallengeAuthenticationPolicy(TokenCredential credential, string scope) : base(credential, new[] { scope })
+        {
+            _scopes = new[] { scope };
+        }
 
         /// <summary>
         /// Creates a new instance of <see cref="ARMChallengeAuthenticationPolicy"/> using provided token credential and scopes to authenticate for.
         /// </summary>
         /// <param name="credential">The token credential to use for authentication.</param>
         /// <param name="scopes">Scopes to authenticate for.</param>
         public ARMChallengeAuthenticationPolicy(TokenCredential credential, IEnumerable<string> scopes)
-            : base(credential, scopes, TimeSpan.FromMinutes(5), TimeSpan.FromSeconds(30)) { }
+            : base(credential, scopes, TimeSpan.FromMinutes(5), TimeSpan.FromSeconds(30))
```

Comment:
Hm, I'm missing something. Where are these `TimeSpan.FromMinutes(5), TimeSpan.FromSeconds(30)` arguments  going. Don't see them on the `BearerTokenAuthenticationPolicy`

Reply:
https://github.com/Azure/azure-sdk-for-net/blob/04b398a6a7ff8fa911b35e459d7e416b7f74b71d/sdk/core/Azure.Core/src/Pipeline/BearerTokenAuthenticationPolicy.cs#L39-L43

Reply:
sdk/core/Azure.Core/src/Shared/ARMChallengeAuthenticationPolicy.cs is a shared file and the ctor you linked to is internal.

It just happens to compile because Azure.Core.Tests have IVTs to Azure.Core

Reply:
I don't think it would work when included in a library.

Reply:
Good catch - this should use the public ctor.